File: modified_scripts/scala/binary_search/deepseek-coder_6.7b.scala

def binarySearch(arr: Array[Int], low: Int, high: Int, key: Int): Boolean = {
    if (high >= low) {
        val mid = low + (high - low) / 2
  
        if (arr(mid) == key) true
        else if (arr(mid) > key) binarySearch(arr, low, mid - 1, key)
        else binarySearch(arr, mid + 1, high, key)
    } else false
}

File: modified_scripts/java/binary_search/deepseek-coder_6.7b.java

public class BinarySearch {
    int binarySearch(int arr[], int low, int high, int x) {
        if (high >= low) {
            int mid = low + (high - low) / 2;
  
            
            if (arr[mid] == x) return mid;
  
             
            if (arr[mid] > x) return binarySearch(arr, low, mid - 1, x);
  
            
            return binarySearch(arr, mid + 1, high, x);
        }
  
        
        return -1;
    }
}
